### ISSUE TITLE
Cosmos dev kit build.sln install error.

### ISSUE THREAD

**Issue by WeRandom** [CosmosOS/Cosmos#1495]
#### Area of Cosmos - What area of Cosmos are we dealing with?
The build.sln part in userkit install.bat

#### Expected Behaviour - What do you think that should happen?
The build.sln shouldn't crash.

#### Actual Behaviour - What unexpectedly happens?
The build - build.sln crashed and showed up with an error (below)

#### Reproduction - How did you get this error to appear?
Double click the userkit install .bat file and let it do its thing.

#### Version - Were you using the User Kit or Dev Kit? And what User Kit version or Dev Kit commit (Cosmos, IL2CPU, X#)?
The dev kit but I think it is the most recent one. Just by the way, I have the dev kit of net 7.1 and I have visual studio and all the required things.
![Screenshot 2020-09-30 095409](https://user-images.githubusercontent.com/66685733/94694493-e8791780-0302-11eb-931a-2bd8b01d63dc.png)

**Comment by WeRandom** [comment]
========================================
    Builder Log
========================================


========================================
    Checking Dependencies
========================================

Repos: IL2CPU, XSharp and Common is installed.
Visual Studio 15.9+ is installed.
Visual Studio Workloads is installed.
Inno Setup is installed.


========================================
    Restore - Build.sln
========================================

"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\MSBuild.exe" "C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build.sln" /nologo /maxcpucount /nodeReuse:False /verbosity:minimal /t:"Restore" /p:DeployExtension=False
  Determining projects to restore...
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.HAL2\Cosmos.HAL2.csproj (in 539 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Deploy.USB\Cosmos.Deploy.USB.csproj (in 539 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.DebugEngine\Cosmos.VS.DebugEngine.csproj (in 540 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.Kernel\Cosmos.Debug.Kernel.csproj (in 434 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Deploy.Pixie\Cosmos.Deploy.Pixie.csproj (in 539 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.System2\Cosmos.System2.csproj (in 540 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.System2_Plugs\Cosmos.System2_Plugs.csproj (in 540 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.Kernel.Plugs.Asm\Cosmos.Debug.Kernel.Plugs.Asm.csproj (in 6 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Core\Cosmos.Core.csproj (in 5 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.Common\Cosmos.Debug.Common.csproj (in 539 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Tests\Cosmos.TestRunner.TestController\Cosmos.TestRunner.TestController.csproj (in 5 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.HyperVServer\Cosmos.Debug.HyperVServer.csproj (in 24 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\TheRingMaster\TheRingMaster.csproj (in 74 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Common\Cosmos.Common.csproj (in 11 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.GDB\Cosmos.Debug.GDB.csproj (in 28 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.Hosts\Cosmos.Debug.Hosts.csproj (in 99 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.Windows\Cosmos.VS.Windows.csproj (in 120 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.DebugConnectors\Cosmos.Debug.DebugConnectors.csproj (in 105 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Build.Tasks\Cosmos.Build.Tasks.csproj (in 116 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.ProjectSystem\Cosmos.VS.ProjectSystem.csproj (in 249 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Core_Plugs\Cosmos.Core_Plugs.csproj (in 276 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Core_Asm\Cosmos.Core_Asm.csproj (in 291 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\IL2CPU\source\Cosmos.IL2CPU\Cosmos.IL2CPU.csproj (in 155 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Tools\NASM\NASM.csproj (in 199 ms).
  Restored C:\Users\Dell\Downloads\Cosmos-master\IL2CPU\source\IL2CPU\IL2CPU.csproj (in 216 ms).
  7 of 32 projects are up-to-date for restore.


========================================
    Build - Build.sln
========================================

"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\MSBuild.exe" "C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build.sln" /nologo /maxcpucount /nodeReuse:False /verbosity:minimal /t:"Build" /p:"OutputPath=C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\\" /p:"VsixOutputPath=C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\\" /p:DeployExtension=False
  Cosmos.Debug.Common -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.Common.dll
DebugSymbolReader.cs(169,30): warning CS0168: The variable 'ex' is declared but never used [C:\Users\Dell\Downloads\Cosmos-master\IL2CPU\source\IL2CPU.Debug.Symbols\IL2CPU.Debug.Symbols.csproj]
DebugInfo.cs(54,24): warning CS0169: The field 'DebugInfo.mDbName' is never used [C:\Users\Dell\Downloads\Cosmos-master\IL2CPU\source\IL2CPU.Debug.Symbols\IL2CPU.Debug.Symbols.csproj]
  IL2CPU.Debug.Symbols -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\IL2CPU.Debug.Symbols.dll
  Cosmos.Debug.Kernel -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.Kernel.dll
  Cosmos.Core.DebugStub -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Core.DebugStub.dll
  Cosmos.Build.Common -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Build.Common.dll
  Cosmos.Debug.DebugConnectors -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.DebugConnectors.dll
  IL2CPU.API -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\IL2CPU.API.dll
  Spruce -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Spruce.dll
  Cosmos.Deploy.Pixie -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Deploy.Pixie.exe
  Cosmos.Debug.HyperVServer -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.HyperVServer.exe
  Cosmos.TestRunner.TestController -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.TestRunner.TestController.dll
  NASM -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\NASM.dll
  VSPropertyPages -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\VSPropertyPages.dll
  XSharp -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\XSharp.dll
  Cosmos.Debug.GDB -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.GDB.exe
  Determining projects to restore...
  Cosmos.Common -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Common.dll
  Cosmos.Deploy.USB -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Deploy.USB.exe
  All projects are up-to-date for restore.
  Cosmos.Debug.HyperVServer -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.HyperVServer.exe
  Cosmos.Debug.HyperVServer -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Debug.Hosts\obj\Debug\netstandard2.0\HyperVServer\
  TheRingMaster -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\TheRingMaster.dll
  Cosmos.Debug.Kernel.Plugs.Asm -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.Kernel.Plugs.Asm.dll
  Cosmos.Debug.Hosts -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Debug.Hosts.dll
  Cosmos.IL2CPU -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.IL2CPU.dll
INTs.cs(235,29): warning CS0649: Field 'INTs.mLastKnownAddress' is never assigned to, and will always have its default value 0 [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Core\Cosmos.Core.csproj]
ACPI.cs(89,30): warning CS0414: The field 'ACPI.SCI_EN' is assigned but its value is never used [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Core\Cosmos.Core.csproj]
ProjectSystem\VS\PlugsProjectTreeProvider.cs(41,44): warning CS0618: 'ProjectTreeProviderBase.TreeUpdateResult.TreeUpdateResult(IProjectTree, bool, IImmutableDictionary<NamedIdentity, IComparable>)' is obsolete: 'LazyFill parameter is no longer supported.' [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.ProjectSystem\Cosmos.VS.ProjectSystem_do5c4rlm_wpftmp.csproj]
ProjectSystem\VS\PlugsProjectTreeProvider.cs(88,44): warning CS0618: 'ProjectTreeProviderBase.TreeUpdateResult.TreeUpdateResult(IProjectTree, bool, IImmutableDictionary<NamedIdentity, IComparable>)' is obsolete: 'LazyFill parameter is no longer supported.' [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.ProjectSystem\Cosmos.VS.ProjectSystem_do5c4rlm_wpftmp.csproj]
ProjectSystem\VS\PropertyPages\OldPropertyManager.cs(27,35): warning CS0067: The event 'OldPropertyManager.ConfigurationsChanged' is never used [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.ProjectSystem\Cosmos.VS.ProjectSystem_do5c4rlm_wpftmp.csproj]
  Cosmos.Core -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Core.dll
  Cosmos.Build.Tasks -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Build.Tasks.dll
  IL2CPU -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\IL2CPU.dll
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(3036,5): error MSB3821: Couldn't process file ProjectSystem\VS\PropertyPages\OldCosmosPropertyPageControl.resx due to its being in the Internet or Restricted zone or having the mark of the web on the file. Remove the mark of the web if you want to process these files. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.ProjectSystem\Cosmos.VS.ProjectSystem.csproj]
  Cosmos.Core_Asm -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Core_Asm.dll
PCIDeviceBridge.cs(13,21): warning CS0108: 'PCIDeviceBridge.SecondaryBusNumber' hides inherited member 'PCIDevice.SecondaryBusNumber'. Use the new keyword if hiding was intended. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.HAL2\Cosmos.HAL2.csproj]
BlockDevice\Ports\Sata.cs(112,21): warning CS0162: Unreachable code detected [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.HAL2\Cosmos.HAL2.csproj]
BlockDevice\Ports\Satapi.cs(105,21): warning CS0162: Unreachable code detected [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.HAL2\Cosmos.HAL2.csproj]
BlockDevice\Ports\Sata.cs(206,21): warning CS0162: Unreachable code detected [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.HAL2\Cosmos.HAL2.csproj]
BlockDevice\Ports\Sata.cs(280,21): warning CS0162: Unreachable code detected [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.HAL2\Cosmos.HAL2.csproj]
Drivers\Video\VGADriver.cs(828,20): warning CS0219: The variable 'values' is assigned but its value is never used [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.HAL2\Cosmos.HAL2.csproj]
  Cosmos.HAL2 -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.HAL2.dll
PCSpeaker.cs(37,24): warning CS1584: XML comment has syntactically incorrect cref attribute 'https://en.wikipedia.org/wiki/Duration_(music)' [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.System2\Cosmos.System2.csproj]
PCSpeaker.cs(52,24): warning CS1584: XML comment has syntactically incorrect cref attribute 'https://en.wikipedia.org/wiki/Musical_note' [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.System2\Cosmos.System2.csproj]
PCSpeaker.cs(53,24): warning CS1584: XML comment has syntactically incorrect cref attribute 'https://en.wikipedia.org/wiki/Audio_frequency' [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.System2\Cosmos.System2.csproj]
Graphics\VBECanvas.cs(84,9): warning CS1587: XML comment is not placed on a valid language element [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.System2\Cosmos.System2.csproj]
  Cosmos.System2 -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.System2.dll
  Cosmos.Core_Plugs -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.Core_Plugs.dll
System\MathImpl.cs(391,26): warning CS0675: Bitwise-or operator used on a sign-extended operand; consider casting to a smaller unsigned type first [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.System2_Plugs\Cosmos.System2_Plugs.csproj]
  Cosmos.System2_Plugs -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.System2_Plugs.dll
CosmosWindowsPackage.cs(67,17): warning VSSDK006: Check whether the result of GetService calls is null. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.Windows\Cosmos.VS.Windows.csproj]
ToolWindows\StackUC.xaml.cs(59,13): warning VSTHRD001: Await JoinableTaskFactory.SwitchToMainThreadAsync() to switch to the UI thread instead of APIs that can deadlock or require specifying a priority. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.Windows\Cosmos.VS.Windows.csproj]
ToolWindows\StackUC.xaml.cs(92,13): warning VSTHRD001: Await JoinableTaskFactory.SwitchToMainThreadAsync() to switch to the UI thread instead of APIs that can deadlock or require specifying a priority. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.Windows\Cosmos.VS.Windows.csproj]
ToolWindows\RegistersControl.xaml.cs(20,13): warning VSTHRD001: Await JoinableTaskFactory.SwitchToMainThreadAsync() to switch to the UI thread instead of APIs that can deadlock or require specifying a priority. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.Windows\Cosmos.VS.Windows.csproj]
ToolWindows\AssemblyUC.xaml.cs(605,13): warning VSTHRD001: Await JoinableTaskFactory.SwitchToMainThreadAsync() to switch to the UI thread instead of APIs that can deadlock or require specifying a priority. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.Windows\Cosmos.VS.Windows.csproj]
  Cosmos.VS.Windows -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.VS.Windows.dll
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(2084,5): warning MSB3277: Found conflicts between different versions of "System.Collections.Immutable" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.DebugEngine\Cosmos.VS.DebugEngine.csproj]
AD7.Impl\AD7Property.cs(229,10): warning CS1030: #warning: 'TODO: String[]' [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.DebugEngine\Cosmos.VS.DebugEngine.csproj]
CosmosDebugEnginePackage.cs(53,42): warning VSSDK006: Check whether the result of GetService calls is null. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.DebugEngine\Cosmos.VS.DebugEngine.csproj]
AD7Util.cs(31,65): warning VSTHRD010: Accessing "ServiceProvider" should only be done on the main thread. Call System.Windows.Threading.Dispatcher.VerifyAccess() first. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.DebugEngine\Cosmos.VS.DebugEngine.csproj]
Commands\DebugExecCommand.cs(40,13): warning VSTHRD010: Accessing "Cosmos.VS.DebugEngine.Commands.DebugExecCommand.DebugExec" should only be done on the main thread. Call System.Windows.Threading.Dispatcher.VerifyAccess() first. [C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.VS.DebugEngine\Cosmos.VS.DebugEngine.csproj]
  Cosmos.VS.DebugEngine -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.VS.DebugEngine.dll
  Cosmos.VS.DebugEngine -> C:\Users\Dell\Downloads\Cosmos-master\Cosmos\Build\VSIP\Cosmos.VS.DebugEngine.vsix


========================================
    Error
========================================

System.Exception: The process failed to execute!
   at Cosmos.Build.Builder.BuildTasks.ProcessBuildTaskBase.WaitForExit(Process process) in C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Build.Builder\BuildTasks\ProcessBuildTaskBase.cs:line 67
   at Cosmos.Build.Builder.BuildTasks.ProcessBuildTaskBase.<>c__DisplayClass5_0.<RunAsync>b__2() in C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Build.Builder\BuildTasks\ProcessBuildTaskBase.cs:line 48
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Cosmos.Build.Builder.ViewModels.MainWindowViewModel.<BuildAsync>d__27.MoveNext() in C:\Users\Dell\Downloads\Cosmos-master\Cosmos\source\Cosmos.Build.Builder\ViewModels\MainWindowViewModel.cs:line 128

**Comment by WeRandom** [comment]
should i uninstall then reinstall
thats what I did and it didn't work

**Comment by quajak** [comment]
This is the problem `error MSB3821: Couldn't process file ProjectSystem\VS\PropertyPages\OldCosmosPropertyPageControl.resx due to its being in the Internet or Restricted zone or having the mark of the web on the file. Remove the mark of the web if you want to process these files.` See #1100

**Comment by WeRandom** [comment]
just so you know the error just changed but ill try fixing what you just posted

**Comment by WeRandom** [comment]
how do you unlock the files

**Comment by WeRandom** [comment]
oh my god it worked thank you so much... there was a button that said unblock. Thank you.

**Comment by WeRandom** [comment]
I didnt want to open a new issue

**Comment by WeRandom** [comment]
in a video it shows assemblyinfo.cs and I cant find it in my project (the video is pretty old)

**Comment by quajak** [comment]
Please open a new issue, since this is a different topic. I would recommend looking for a newer tutorial, the wiki and https://cosmosos.github.io/articles/Kernel/CGS.html has some (mostly up to date) information on the topic.
